### ISSUE TITLE
Bug: Cannot set title with emoji symbol

### ISSUE THREAD

**Issue by oluckyman** [TortugaPower/BookPlayer#999]
## Environment

 - OS Version: 16.6
 - App Version: 5.0.8

## Description
In Edit screen when I try to save title with emoji symbol, it shows error with cryptic message and does not save my changes.

> Error: "Old folder name" couldn't be moved to "Processed" because either the former doesn't exist, or the folder containing the latter doesn't exist.

 <img width="408" alt="image" src="https://github.com/TortugaPower/BookPlayer/assets/642673/6e79f396-c3cf-4be8-a376-a067efca3eeb">

## Steps to reproduce the behavior
1. Swipe left a folder
2. Select Options…
3. Select Details
4. Add emoji symbol into Title field
5. Tap Save
…
It shows error and does not allow to save changes.

**Comment by harrisi** [comment]
This seems to work fine for me, so I don't think it's just the emoji in the title.

Running BookPlayer 5.0.8-7, iOS 16.6. 

![IMG_1961](https://github.com/TortugaPower/BookPlayer/assets/14868079/eab5e1ea-f36f-459c-af0b-eced9256e8c6)

**Comment by GianniCarlo** [comment]
Yeah this looks like a problem with the backing folder on disk, as renaming a folder attempts to move the original folder 'Archive' to the new destination 'Archive 📚' and fails at this point

<img width="566" alt="Screenshot 2023-09-05 at 16 09 27" src="https://github.com/TortugaPower/BookPlayer/assets/14112819/7fa055e8-5538-46ce-a080-c2f423fe4536">

Can you check on the app Files → On my iPhone → BookPlayer → Processed, if the folder is there? if for some reason it's not there, you can create it manually there, and on the app it should allow you to rename it, but I'm curious on how the backing folder disappeared, could you provide more context on the library hierarchy? and if this is on a secondary device that is using the sync feature? because when the app fetches new items from the server, it should create the backing folder(s) as well 🤔 (so probably something to do more tests on, and see if there's a bug in there)

**Comment by oluckyman** [comment]
Thanks for the quick response!

`Processed` folder exists. And has 3 books inside (but the library has more books).
I have only one device with BookPlayer.
The library hierarchy is almost flat. I move finished books into Archive folder, so it has the mix of single-file books, folder-with-files books and combined-into-volume books. And also, I uploaded artwork for Archive folder.

Initially I thought the problem is with emoji symbol, but now I tried to rename title without emoji and got the same error.

**Comment by oluckyman** [comment]
Just realized that the 3 books in the `Processed` folder are all that remain from my library of 22 books.
If I tap on any of the 19 missing books, it shows the error,
_"File missing! This book's file was removed from your device. Import the file again to play the book ..."_
Something is very wrong if the content can just disappear. It will be a pain to search for and reimport these books again. I probably will not.

**Comment by GianniCarlo** [comment]
@oluckyman yes, I meant to follow up about this, I have three follow up questions, as the books shouldn't have disappeared:
* This is a long shot, but by any chance were you a user back on version 3.0.0? there was a migration to v4 where we reorganized `Processed` folder contents, to have a 1:1 parity with what's displayed on the app (before, the imported files had a hashed name, and were all at the same level)
* We also had a bug that was solved this year, where if you selected one of the items from the Files app inside the Processed folder, it would open the app, and prompt to import again, and when cancelling, it would erase the source file (thinking it was a copy)
* The other thing is if you recently switched phones, and a iCloud backup was used, probably the backing files weren't included in there (which if you have it enabled in Settings, then it's definitely something we can look into)

Let me know if you remember anything else in particular, as the move/delete operations on the app will throw and show an error if the operation fails

<img width="538" alt="Screenshot 2023-09-08 at 21 02 07" src="https://github.com/TortugaPower/BookPlayer/assets/14112819/60cc92c6-7b66-4635-8419-91668813df92">

**Comment by oluckyman** [comment]
I've been a longtime user, but not sure if I remember v3.
The missing books were there last week.
The only thing I did the day I opened this issue was import a few books and try to rename the Archive folder.
Also, there was an app crash when I tried to play one of the new books (in m4b format). But after restarting the app and reimporting the book file into a dedicated folder, it was fine.
I don't remember migrating from version 3 to 4. I didn't open the Processed folder in Files until I opened this issue. I haven't switched phones or used a backup.
